src/session_impl.cpp

```diff
@@ -5524,13 +5527,16 @@ namespace {
 	std::uint16_t session_impl::listen_port(listen_socket_t* sock) const
 	{
 		if (m_listen_sockets.empty()) return 0;
-		if (sock)
-		{
-			// if we're using a proxy, we won't be able to accept any TCP
-			// connections. Not even uTP connections via the port we know about.
+		if (sock){
+
+			// if we're using a proxy, the user should be able to choose
+			// if we listen for incoming connections or not based on accept_incoming flag.
```

Comment:
```suggestion
			// if we're using a proxy, whether we accept incoming connections on the local listen port
			// or not is determined by the proxy_accept_incoming flag.
```

Expressing an opinion without clarifying the facts, like "the user should be able to" isn't very helpful here. Is the user able to or not?